stdlib/public/core/BridgeObjectiveC.swift

```diff
@@ -369,7 +369,7 @@ public struct AutoreleasingUnsafeMutablePointer<Pointee /* TODO : class */>
     /// Retrieve the value the pointer points to.
     @_transparent get {
       // We can do a strong load normally.
-      return unsafeBitCast(self, to: UnsafeMutablePointer<Pointee>.self).pointee
+      return UnsafeMutablePointer(mutating: self).pointee
```

Comment:
It may have something to do with `UnsafeMutablePointer` wrapping its `pointee` into an `UnsafePointer`, whereas immutable pointer does not perform that copy. But then this should be equivalent to just `return UnsafePointer(self).pointee`.